File: frontend/types/schools.ts
import type { SourceAttribution } from "./common";

export interface OfstedInspectionSummary {
  status: "matched" | "data_unavailable";
  urn: string | null;
  most_recent_category_of_concern: string | null;
  full_inspection_type: string | null;
  full_inspection_start_date: string | null;
  full_inspection_publication_date: string | null;
  safeguarding_standards: string | null;
  inclusion: string | null;
  curriculum_and_teaching: string | null;
  achievement: string | null;
  attendance_and_behaviour: string | null;
  personal_development_and_wellbeing: string | null;
  early_years: string | null;
  post_16_provision: string | null;
  leadership_and_governance: string | null;
  oeif_start_date: string | null;
  oeif_publication_date: string | null;
  oeif_overall_effectiveness: string | null;
  oeif_safeguarding_effective: boolean | null;
  ungraded_inspection_date: string | null;
  ungraded_publication_date: string | null;
  ungraded_overall_outcome: string | null;
  source: SourceAttribution | null;
  message: string | null;
}

export interface SchoolSearchResult {
  urn: string;
  establishment_name: string;
  la_name: string | null;
  type_name: string | null;
  phase_name: string | null;
  status_name: string | null;
  postcode: string | null;
  town: string | null;
  source: SourceAttribution;
}

export interface SchoolSearchResponse {
  query: string;
  results: SchoolSearchResult[];
  total: number;
  dataset_version: string | null;
  message: string | null;
  suggestions: SchoolSearchResult[];
}

export interface SchoolDetail {
  urn: string;
  establishment_name: string;
  la_name: string | null;
  type_name: string | null;
  type_group_name: string | null;
  status_name: string | null;
  phase_name: string | null;
  statutory_low_age: number | null;
  statutory_high_age: number | null;
  gender_name: string | null;
  religious_character_name: string | null;
  school_capacity: number | null;
  number_of_pupils: number | null;
  ukprn: string | null;
  open_date: string | null;
  close_date: string | null;
  street: string | null;
  locality: string | null;
  town: string | null;
  county_name: string | null;
  postcode: string | null;
  website: string | null;
  telephone: string | null;
  head_first_name: string | null;
  head_last_name: string | null;
  region_name: string | null;
  country_name: string | null;
  source: SourceAttribution;
  ofsted: OfstedInspectionSummary;
  limitations: string[];
}

File: backend/app/schemas/schools.py
from __future__ import annotations

from datetime import date
from typing import Literal

from pydantic import BaseModel, Field

from app.schemas.common import SourceAttribution


class OfstedInspectionSummary(BaseModel):
    status: Literal["matched", "data_unavailable"]
    urn: str | None = None
    most_recent_category_of_concern: str | None = None
    full_inspection_type: str | None = None
    full_inspection_start_date: date | None = None
    full_inspection_publication_date: date | None = None
    safeguarding_standards: str | None = None
    inclusion: str | None = None
    curriculum_and_teaching: str | None = None
    achievement: str | None = None
    attendance_and_behaviour: str | None = None
    personal_development_and_wellbeing: str | None = None
    early_years: str | None = None
    post_16_provision: str | None = None
    leadership_and_governance: str | None = None
    oeif_start_date: date | None = None
    oeif_publication_date: date | None = None
    oeif_overall_effectiveness: str | None = None
    oeif_safeguarding_effective: bool | None = None
    ungraded_inspection_date: date | None = None
    ungraded_publication_date: date | None = None
    ungraded_overall_outcome: str | None = None
    source: SourceAttribution | None = None
    message: str | None = None


class SchoolSearchResult(BaseModel):
    urn: str
    establishment_name: str
    la_name: str | None = None
    type_name: str | None = None
    phase_name: str | None = None
    status_name: str | None = None
    postcode: str | None = None
    town: str | None = None
    source: SourceAttribution


class SchoolSearchResponse(BaseModel):
    query: str
    results: list[SchoolSearchResult]
    total: int
    dataset_version: str | None = None
    message: str | None = None
    suggestions: list[SchoolSearchResult] = Field(default_factory=list)


class SchoolDetail(BaseModel):
    urn: str
    establishment_name: str
    la_name: str | None = None
    type_name: str | None = None
    type_group_name: str | None = None
    status_name: str | None = None
    phase_name: str | None = None
    statutory_low_age: int | None = None
    statutory_high_age: int | None = None
    gender_name: str | None = None
    religious_character_name: str | None = None
    school_capacity: int | None = None
    number_of_pupils: int | None = None
    ukprn: str | None = None
    open_date: date | None = None
    close_date: date | None = None
    street: str | None = None
    locality: str | None = None
    town: str | None = None
    county_name: str | None = None
    postcode: str | None = None
    website: str | None = None
    telephone: str | None = None
    head_first_name: str | None = None
    head_last_name: str | None = None
    region_name: str | None = None
    country_name: str | None = None
    source: SourceAttribution
    ofsted: OfstedInspectionSummary
    limitations: list[str] = Field(default_factory=list)
